src/extensionConsole/pages/settings/GeneralSettings.tsx

```diff
@@ -24,10 +24,23 @@ import { useDispatch, useSelector } from "react-redux";
 import { selectSettings } from "@/store/settingsSelectors";
 import reportEvent from "@/telemetry/reportEvent";
 import { Events } from "@/telemetry/events";
+import { useGetTheme } from "@/hooks/useTheme";
+import { selectTelemetryOrganizationId } from "@/auth/authSelectors";
 
 const GeneralSettings: React.FunctionComponent = () => {
   const dispatch = useDispatch();
   const { isFloatingActionButtonEnabled } = useSelector(selectSettings);
+
+  const theme = useGetTheme();
+
+  const telemetryOrganizationId = useSelector(selectTelemetryOrganizationId);
+
+  // Disable FAB for enterprise and partner users
+  const disableFloatingActionButton =
+    Boolean(telemetryOrganizationId) || theme !== "default";
+
+  const checked = isFloatingActionButtonEnabled && !disableFloatingActionButton;
```

Comment:
nit: my brain initially read "checked" as "something was checked (as in validated)"
```suggestion
  const enabled = isFloatingActionButtonEnabled && !disableFloatingActionButton;
```